flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/TableSinkITCase.scala

```diff
@@ -192,4 +193,41 @@ class TableSinkITCase(mode: StateBackendMode) extends StreamingWithStateTestBase
     )
     assertEquals(expected.sorted, result.sorted)
   }
+
+  @Test
+  def testCreateTableAsSelect(): Unit = {
+    tEnv
+      .executeSql("""
+                    |CREATE TABLE MyCtasTable
+                    | WITH (
+                    |   'connector' = 'values',
+                    |   'sink-insert-only' = 'true'
+                    |) AS
+                    |  SELECT
+                    |    `person`,
+                    |    `votes`
+                    |  FROM
+                    |    src
+                    |""".stripMargin)
+      .await()
+    val actual = TestValuesTableFactory.getResults("MyCtasTable")
+    val expected = List(
+      "+I[jason, 1]",
+      "+I[jason, 1]",
+      "+I[jason, 1]",
+      "+I[jason, 1]"
+    )
+    Assertions.assertThat(actual.sorted).isEqualTo(expected.sorted)
+  }
+
+  @Test
+  def testCreateTableAsSelectWithoutOptions(): Unit = {
+    // TODO CTAS supports ManagedTable
+    Assertions
```

Comment:
Please also explain the reason?

Reply:
If the connector option is not specified, Flink will creates a Managed table.
Managed table relies on checkpoint to commit and the data is visible only after commit.
Managed table requires two layers of log storage and file storage and depends on the flink table store, CTAS will support Managed Table in the future.